sw/device/lib/testing/test_framework/check.h

```diff
@@ -33,7 +33,9 @@
  */
 #define CHECK_IMPL(condition, ...)                   \
   ({                                                 \
-    if (!(condition)) {                              \
+    status_t sts_ = OK_STATUS();                     \
+    volatile bool res_ = (condition);                \
+    if (!res_) {                                     \
```

Comment:
Why does `condition` have to be assigned to a `volatile bool` variable and cannot directly be used as condition to `if`?

Reply:
Interesting, thanks! But in that case, doesn't the compiler do the correct thing and replaces the `CHECK` macro invocation with either `INTERNAL()` or `OK_STATUS()`?

Reply:
Ah okay, that makes sense. I suggest you add a comment describing why `volatile bool` is necessary, so it does not get removed by mistake